flink-end-to-end-tests/flink-stream-sql-test/src/test/java/org/apache/flink/sql/tests/StreamSQLTestProgramScalaPlannerITCase.java

```diff
@@ -0,0 +1,90 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.sql.tests;
+
+import org.apache.flink.connector.testframe.container.FlinkContainers;
+import org.apache.flink.connector.testframe.container.FlinkContainersSettings;
+import org.apache.flink.connector.testframe.container.TestcontainersSettings;
+
+import org.junit.jupiter.api.AfterAll;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class StreamSQLTestProgramScalaPlannerITCase extends AbstractStreamSQLTestProgramITCase {
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(StreamSQLTestProgramScalaPlannerITCase.class);
+    private static final String DIST_DIR = System.getProperty("distDir");
+
+    @Override
+    protected FlinkContainers createFlinkContainers() {
+        // Swap planner jar files in `lib/` and `opt/` folders
+        swapPlannerLoaderWithPlannerScala();
+
+        return FlinkContainers.builder()
+                .withFlinkContainersSettings(
+                        FlinkContainersSettings.builder().numTaskManagers(4).build())
+                .withTestcontainersSettings(
+                        TestcontainersSettings.builder().network(NETWORK).logger(LOGGER).build())
+                .build();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        swapPlannerScalaWithPlannerLoader();
+    }
+
+    private void swapPlannerLoaderWithPlannerScala() {
+        move("flink-table-planner-loader", "lib", "opt");
+        move("flink-table-planner_", "opt", "lib");
+    }
+
+    private static void swapPlannerScalaWithPlannerLoader() {
+        move("flink-table-planner-loader", "opt", "lib");
+        move("flink-table-planner_", "lib", "opt");
+    }
+
+    private static void move(final String filePattern, final String from, final String to) {
+        final Path fromDirectory = Paths.get(DIST_DIR, from);
+        try (final Stream<Path> filesStream = Files.list(fromDirectory)) {
+            final List<Path> files =
+                    filesStream
+                            .filter(entry -> !Files.isDirectory(entry))
+                            .filter(file -> file.toString().endsWith(".jar"))
+                            .filter(file -> file.getFileName().toString().startsWith(filePattern))
+                            .collect(Collectors.toList());
+            if (files.size() != 1) {
+                throw new IllegalStateException(
+                        "Found multiple file pattern '" + filePattern + "', expected only one.");
```

Comment:
It can also contain zero patterns no? 
Maybe we can improve the error message sth like `"Found " + file.size() " patterms, expected only one."`?